dragonfly/player/player.go

```diff
@@ -1100,6 +1100,12 @@ func (p *Player) StartBreaking(pos world.BlockPos, face world.Face) {
 	ctx := event.C()
 	p.handler().HandleStartBreak(ctx, pos)
 	ctx.Continue(func() {
+		if punchable, ok := p.World().Block(pos).(block.Punchable); ok {
+			p.swingArm()
+			punchable.Punch(pos, face, p.World(), p)
+			return
+		}
+
```

Comment:
This might prevent breaking punchable blocks. I don't think you'd want to return here.